### PR TITLE
updates for task dependency actions

### PR DESCRIPTION
**Pull request by tamram**

### CODE REVIEW

File: CSharp/ArticleProjects/TaskDependencies/Program.cs [hunk 1]
```diff
@@ -21,15 +18,14 @@ public static void Main(string[] args)
             {
                 // Call the asynchronous version of the Main() method. This is done so that we can await various
                 // calls to async methods within the "Main" method of this console application.
-                MainAsync(args).Wait();
+                MainAsync(args).GetAwaiter().GetResult();
             }
-            catch (AggregateException ae)
+            catch (Exception e)
```
Comment (itowlson) [inline CSharp/ArticleProjects/TaskDependencies/Program.cs]: As this is no longer AggregateException, we should change the "One or more exceptions" language below.

File: CSharp/ArticleProjects/TaskDependencies/Program.cs [hunk 2]
```diff
@@ -58,77 +54,91 @@ public static void Main(string[] args)
...
+                        // specify string representations of integers as their ids.
+                        new CloudTask("1", "cmd.exe /c echo 1"),
+                        new CloudTask("2", "cmd.exe /c echo 2"),
+                        new CloudTask("3", "cmd.exe /c echo 3"),
+
+                        // Task A is the parent task.
+                        new CloudTask("A", "cmd.exe /c echo A")
+                        {
+                            // Specify exit conditions for task A and their dependency actions.
+                            ExitConditions = new ExitConditions()
```
Comment (itowlson) [inline CSharp/ArticleProjects/TaskDependencies/Program.cs]: Do not use `()` in the parameterless constructor call when supplying an initialiser list.  (Same comment throughout this code block.)

File: CSharp/ArticleProjects/TaskDependencies/Program.cs [hunk 3]
```diff
@@ -137,24 +147,34 @@ public static void Main(string[] args)
                     Console.WriteLine("All tasks completed successfully.");
                     Console.WriteLine();
                 }
-                catch (TimeoutException e)
+            }
+            catch (Exception e)
+            {
+                Console.WriteLine();
+                Console.WriteLine("One or more exceptions occurred.");
```
Comment (itowlson) [inline CSharp/ArticleProjects/TaskDependencies/Program.cs]: Again "One or more" is probably misleading.

**Comment by tamram** [comment]
@itowlson - thanks!

**Pull request merged**